From e71854bb2710d2080a485321c32121185260a1ec Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Sat, 30 Dec 2017 16:03:20 +0300
Subject: [PATCH 2/2] lineage: Enable boot and system server dex-preopt

As stated in https://android.googlesource.com/platform/build/+/418258c
this is required in O-MR1 since sepolicy now prevents system server from
loading anything from /data making our system extremely slow and janky.

Change-Id: I215ec7ebe95981ef0b953fe938e12b83675795ab
---
 config/BoardConfigLineage.mk | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/config/BoardConfigLineage.mk b/config/BoardConfigLineage.mk
index 0336314..d5ec409 100644
--- a/config/BoardConfigLineage.mk
+++ b/config/BoardConfigLineage.mk
@@ -2,3 +2,13 @@
 ifeq ($(WITH_LINEAGE_CHARGER),true)
     BOARD_HAL_STATIC_LIBRARIES := libhealthd.lineage
 endif
+
+# Dexpreopt
+ifeq ($(HOST_OS),linux)
+  ifneq ($(TARGET_BUILD_VARIANT),eng)
+    ifeq ($(WITH_DEXPREOPT),)
+      WITH_DEXPREOPT := true
+      WITH_DEXPREOPT_BOOT_IMG_AND_SYSTEM_SERVER_ONLY := true
+    endif
+  endif
+endif
-- 
2.7.4

